app/questionnaire/questionnaire_schema.py

```diff
@@ -285,14 +282,45 @@ def _populate_answer_dependencies(self) -> None:
                                 option["detail_answer"], block_id=block["id"]
                             )
 
-    def _update_answer_dependencies_for_calculated_summary(
-        self, calculated_summary_answer_ids: Iterable[str], block_id: str
+    def _update_answer_dependencies_for_summary(self, block: ImmutableDict) -> None:
+        if block["type"] == "CalculatedSummary":
+            self._update_answer_dependencies_for_calculated_summary_dependency(
+                calculated_summary_block=block, dependent_block=block
+            )
+        else:  # must be GrandCalculatedSummary
```

Comment:
I think it'd be worth guarding against misuse of this function by 
`if block["type"] == "GrandCalculatedSummary":`
and having nothing else happen if it's something else, or even throw an exception but do nothing seems more in line with rest of the QuestionnaireSchema.

Reply:
happy to make that change, safe and difference is negligible